From 735c7730d70d3b5610e92381944a862ecfa4f0b9 Mon Sep 17 00:00:00 2001
From: Sergei Makarov <serg.makarov@samsung.com>
Date: Mon, 10 Apr 2023 18:45:09 +0300
Subject: [PATCH 15/29] [pytorch, secndp] Enable shared memory usage

AxDIMM now uses shared memory for simulation which requires
additional setup from Gramine part. Gramine does not allow
to mount individual files from the host, so we mount `/dev/shm`
entirely. In the future this should be replaced with safer
alternative. For example, we can set up needed devices on
application start.

Related to MCS23-722

Signed-off-by: Sergei Makarov <serg.makarov@samsung.com>
---
 application/axdimm_secndp_test/secndp.manifest.template | 4 +++-
 application/pytorch/conf.pytorch.manifest.template      | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/application/axdimm_secndp_test/secndp.manifest.template b/application/axdimm_secndp_test/secndp.manifest.template
index 76853f65..4671e21b 100644
--- a/application/axdimm_secndp_test/secndp.manifest.template
+++ b/application/axdimm_secndp_test/secndp.manifest.template
@@ -24,6 +24,7 @@ fs.mounts = [
   { uri = "file:{{ axdimm_driver }}/build/secure", path = "{{ axdimm_driver }}/build/secure" },
   { uri = "file:/sys/bus/", path = "/sys/bus/", type="chroot" }, #unsafe but we can't mount single file from sysfs
   { type = "tmpfs", path = "/tmp" }
+  { type = "shm" },
 ]
 
 sgx.nonpie_binary = true
@@ -33,7 +34,8 @@ sgx.debug = true
 sgx.insecure__enable_shared_range = "true"
 
 sgx.allowed_files = [
-  "file:/sys/bus/dax/drivers/device_dax/dax0.0/ranks/"
+  "file:/sys/bus/dax/drivers/device_dax/dax0.0/ranks/",
+  "file:/dev/shm/" # We need to allow axdimm device to enable write access to it
 ]
 
 sgx.trusted_files = [
diff --git a/application/pytorch/conf.pytorch.manifest.template b/application/pytorch/conf.pytorch.manifest.template
index dbcadbe1..edc87f83 100644
--- a/application/pytorch/conf.pytorch.manifest.template
+++ b/application/pytorch/conf.pytorch.manifest.template
@@ -24,6 +24,7 @@ fs.mounts = [
   { uri = "file:{{ axdimm_driver }}/build/secure", path = "{{ axdimm_driver }}/build/secure"},
   { uri = "file:{{ dlrm_data_path }}", path = "/input" },
   { type = "tmpfs", path = "/tmp" },
+  { type = "shm" },
   { uri = "dev:/dev/axdimm_device0", path = "/dev/axdimm_device0" },
   { uri = "file:/sys/bus/", path = "/sys/bus/", type="chroot" }, #unsafe but we can't mount single file from sysfs
 ]
@@ -50,6 +51,7 @@ sgx.trusted_files = [
 ]
 
 sgx.allowed_files = [
+  "file:/dev/shm/", # We need to allow axdimm device to enable write access to it
   "file:./run_kaggle_pt/",
   "file:/sys/bus/dax/drivers/device_dax/dax0.0/ranks/",
   "file:{{ dlrm_data_path }}",
-- 
2.34.1

